src/base/bittorrent/session.cpp

```diff
@@ -4218,9 +4180,38 @@ void Session::setKeepTorrentTopLevelFolder(const bool value)
 // Read alerts sent by the BitTorrent session
 void Session::readAlerts()
 {
-    const std::vector<lt::alert *> alerts = getPendingAlerts();
-    for (const lt::alert *a : alerts)
-        handleAlert(a);
+    if (m_nextAlert == m_alerts.end()) {
+        m_alerts = getPendingAlerts();
+        m_nextAlert = m_alerts.begin();
+        handleNextAlert();
+    }
+    else {
+        m_hasPendingAlerts = true;
+    }
+}
+
+void Session::handleNextAlert()
+{
+    Q_ASSERT(m_nextAlert != m_alerts.end());
+
+    QElapsedTimer timer;
+    timer.start();
+    while (m_nextAlert != m_alerts.end() && !timer.hasExpired(10)) {
+        handleAlert(*(m_nextAlert++));
```

Comment:
I think this way you won't process the very first alert in m_alerts.

Reply:
I think you're wrong.

Reply:
Note that this is a **post** increment.

Reply:
IDK, it seems at first m_nextAlert is m_alerts.begin();, but here you're incrementing m_nextAlert before processing it.

Reply:
`*m_nextAlert++` instead of `*(m_nextAlert++)` ??

Reply:
oopsie, that bracket is confusing, Sorry.

Reply:
Without parentheses, it looks like the increment is applied to the dereferenced value... Matter of taste.